static/app/views/organizationGroupDetails/allEventsTable.tsx

```diff
@@ -99,4 +65,88 @@ const AllEventsTable = (props: Props) => {
   );
 };
 
+type ColumnInfo = {columnTitles: string[]; fields: string[]};
+
+const getColumns = (group: Group, organization: Organization): ColumnInfo => {
+  const isPerfIssue = group.issueCategory === IssueCategory.PERFORMANCE;
+  const isReplayEnabled = organization.features.includes('session-replay-ui');
+
+  const {fields: platformSpecificFields, columnTitles: platformSpecificColumnTitles} =
+    getPlatformColumns(group.project.platform ?? group.platform, {isReplayEnabled});
+
+  const fields: string[] = [
+    'id',
+    'transaction',
+    'title',
+    'release',
+    'environment',
+    'user.display',
+    'device',
+    'os',
+    ...platformSpecificFields,
+    ...(isPerfIssue ? ['transaction.duration'] : []),
+    'timestamp',
+  ];
+
+  const columnTitles: string[] = [
+    t('event id'),
+    t('transaction'),
+    t('title'),
+    t('release'),
+    t('environment'),
+    t('user'),
+    t('device'),
+    t('os'),
+    ...platformSpecificColumnTitles,
+    ...(isPerfIssue ? [t('total duration')] : []),
+    t('timestamp'),
+    t('minidump'),
+  ];
+
+  return {
+    fields,
+    columnTitles,
+  };
+};
+
+const getPlatformColumns = (
+  platform: PlatformKey | undefined,
+  options: {isReplayEnabled: boolean}
+): ColumnInfo => {
+  const replayField = options.isReplayEnabled ? ['replayId'] : [];
+  const replayColumnTitle = options.isReplayEnabled ? [t('replay')] : [];
+
+  const categoryToColumnMap: {[key: string]: ColumnInfo} = {
+    [PlatformCategory.FRONTEND]: {
+      fields: ['url', 'browser', ...replayField],
+      columnTitles: [t('url'), t('browser'), ...replayColumnTitle],
+    },
+    [PlatformCategory.BACKEND]: {
+      fields: ['url', 'runtime'],
+      columnTitles: [t('url'), t('runtime')],
+    },
+    [PlatformCategory.MOBILE]: {
+      fields: ['url'],
+      columnTitles: [t('url')],
+    },
+    [PlatformCategory.DESKTOP]: {
+      fields: ['os'],
+      columnTitles: [t('os')],
+    },
+    [PlatformCategory.OTHER]: {
+      fields: [],
+      columnTitles: [],
+    },
+  };
+
+  let platformCategory = platformToCategory(platform);
+
+  // backend end serverless have the same columns
+  if (platformCategory === PlatformCategory.SERVERLESS) {
+    platformCategory = PlatformCategory.BACKEND;
+  }
```

Comment:
Instead of this, I think we should just create another entry for `PlatformCategory.SERVERLESS` within `categoryToColumnMap`. It will have the same columns but I think it makes more sense to keep the structure of passing the `platformCategory` to the map instead of having some intermediary logic that re-assigns the value

Reply:
I do agree the reassignment is not ideal,

We want to ensure serverless and backend always get the same columns, even down the line as we modify columns, to prevent accidental changes I'd prefer not duplicating. How about adding this
`categoryToColumnMap[PlatformCategory.SERVERLESS] = categoryToColumnMap[PlatformCategory.BACKEND]`?

Reply:
Yeah, lets do that, thanks @0Calories !
That solves that typing issue I just mentioned, with mising '3'